File: app/src/main/java/com/example/myfakequotesapp/View/FilmResults.java
package com.example.myfakequotesapp.View;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Toast;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.Observer;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.example.myfakequotesapp.Model.Quote;
import com.example.myfakequotesapp.R;
import com.example.myfakequotesapp.View.Adapters.FilmResultsAdapter;
import com.example.myfakequotesapp.ViewModel.FilmResultsViewModel;
import com.example.myfakequotesapp.api.ApiFactory;
import com.example.myfakequotesapp.api.ApiService;
import com.example.myfakequotesapp.databinding.ActivityFilmResultsBinding;
import com.example.myfakequotesapp.pojo.Example;
import com.example.myfakequotesapp.pojo.Result;

import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

import io.reactivex.Scheduler;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.disposables.Disposable;
import io.reactivex.functions.Consumer;
import io.reactivex.schedulers.Schedulers;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

import static android.content.ContentValues.TAG;

public class FilmResults extends AppCompatActivity {
    private ActivityFilmResultsBinding binding;
    private String filmTitle;
    private FilmResultsAdapter adapter;
    private RecyclerView recyclerViewPosters;
    private FilmResultsViewModel filmResultsViewModel;



    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        // биндим и размещаем интерфейс в активити
        binding = ActivityFilmResultsBinding.inflate(getLayoutInflater());
        View view = binding.getRoot();
        setContentView(view);

        Intent intent = getIntent();
        filmTitle = intent.getStringExtra("filmTitle");

        initRecyclerView();
    }

    private void initRecyclerView(){
        recyclerViewPosters = findViewById(R.id.recyclerViewPosters);
        recyclerViewPosters.setLayoutManager(new GridLayoutManager(this, 2));

        adapter = new FilmResultsAdapter();
        recyclerViewPosters.setAdapter(adapter);
        filmResultsViewModel = new FilmResultsViewModel();

        filmResultsViewModel.getExample(filmTitle).observe(this, new Observer<Example>() {
            @Override
            public void onChanged(Example example) {
                adapter.setFilms(example.getResults());
            }
        });
    }
}

File: app/src/main/java/com/example/myfakequotesapp/View/FilmResults.kt
package com.example.myfakequotesapp.View

import androidx.recyclerview.widget.RecyclerView
import com.example.myfakequotesapp.R
import com.example.myfakequotesapp.View.Adapters.FilmResultsAdapter.OnPosterClickListener
import androidx.appcompat.app.AppCompatActivity
import com.example.myfakequotesapp.View.Adapters.FilmResultsAdapter
import com.example.myfakequotesapp.ViewModel.FilmResultsViewModel
import android.os.Bundle
import android.content.Intent
import androidx.recyclerview.widget.GridLayoutManager
import android.view.View
import com.example.myfakequotesapp.databinding.ActivityFilmResultsBinding

class FilmResults : AppCompatActivity() {
    private var binding: ActivityFilmResultsBinding? = null
    private var filmTitle: String? = null
    private var adapter: FilmResultsAdapter? = null
    private var recyclerViewPosters: RecyclerView? = null
    private var filmResultsViewModel: FilmResultsViewModel? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // биндим и размещаем интерфейс в активити
        binding = ActivityFilmResultsBinding.inflate(
            layoutInflater
        )
        val view: View = binding!!.root
        setContentView(view)
        val intent = intent
        filmTitle = intent.getStringExtra("filmTitle")
        initRecyclerView()
    }

    private fun initRecyclerView() {
        recyclerViewPosters = findViewById(R.id.recyclerViewPosters)
        recyclerViewPosters?.setLayoutManager(GridLayoutManager(this, 2))
        adapter = FilmResultsAdapter()
        recyclerViewPosters?.setAdapter(adapter)
        filmResultsViewModel = FilmResultsViewModel()
        filmResultsViewModel!!.getExample(filmTitle)
            .observe(this) { example -> adapter!!.setFilms(example?.results) }
        adapter?.setOnPosterClickListener(object : OnPosterClickListener {
            override fun onPosterClick(position: Int) {
                val film = adapter!!.getFilms()!!.get(position)
                val intent = Intent(this@FilmResults, AddQuoteActivity::class.java)
                intent.putExtra("title", film?.title)
                startActivity(intent)
            }
        })
    }
}